python/ray/tests/test_network_failures.py

```diff
@@ -0,0 +1,56 @@
+import os
+import pytest
+import sys
+
+import ray
+from ray._private.test_utils import Cgroup2NetworkBlocker
+from ray.exceptions import ActorUnavailableError
+
+pytest.skipif(sys.platform == "linux", "Cgroup2NetworkBlocker only works on Linux.")
+
+
+def test_network_blocker_works():
+    import requests
+
+    with Cgroup2NetworkBlocker(os.getpid()):
+        with pytest.raises(requests.exceptions.RequestException):
+            print(requests.get("http://example.com", timeout=5))
+    print(requests.get("http://example.com", timeout=5))
+
+
+@pytest.skipif(sys.platform == "linux", "Cgroup2NetworkBlocker only works on Linux.")
+def test_actor_unavailable(ray_start_cluster):
```

Comment:
Can you add a test with 2 actors A and B that call each other? The test should:

1. Start A and B.
2. Make A call B, and check that the call succeeds.
3. Block A's outgoing traffic.
4. Make A call B and check that you get an ActorUnavailableError.

Reply:
And can you add a similar test for Ray tasks (e.g. a Ray actor runs a task, tries to `ray.get` the task, and then gets an appropriate error). If we haven't handled this edge case for Ray tasks yet, then we can defer this for a future PR when we do.